### PR TITLE
[12_5_X] [L1T] Backport of: Phase-2, include missing collections in outputs

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

This PR addresses the missing Tk object collections in the CMSSW_12_5_0_pre5 RelVals. It includes fixes were needed to avoid errors after adding in the objects to the EventContent. One error that came up in the FEVTDEBUGHLToutput_step was: The class "l1t::TkJetWord" is compiled and for its data member "tkJetWord_" we do not have a dictionary for the collection "bitset<70>". Because of this, we will not be able to read or write this data member.
classes_def.xml was adjusted to avoid the error. Local PR: https://github.com/cms-l1t-offline/cmssw/pull/1048

This PR also adds missing collections for Phase 2 tau-jet-met to the event content. Specifically the 9x9 Sums for histogrammed jets, the NN Puppi taus, and the HPS taus. Local PR: https://github.com/cms-l1t-offline/cmssw/pull/1047

We need this PR in 12_5 for our Phase-2 L1 MC production

Backport of https://github.com/cms-sw/cmssw/pull/39401 and https://github.com/cms-sw/cmssw/pull/39499

### CODE REVIEW

**Comment by cecilecaillol** [comment]
backport of https://github.com/cms-sw/cmssw/pull/39401

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by perrotta** [comment]
@cecilecaillol @BenjaminRS I assume all these collections are needed, and therefore the master PR had just been signed. Nonetheless, in this backport PR could you please add an estimate of the increase of the event size, possibly already in the PR description?

**Comment by perrotta** [comment]
This PR backports both https://github.com/cms-sw/cmssw/pull/39401 and https://github.com/cms-sw/cmssw/pull/39499

**Comment by srimanob** [comment]
+Upgrade

Backport PR. Waiting for size estimation for additional collections.

**Comment by srimanob** [comment]
Note to @perrotta @rappoccio 
With this PR together with https://github.com/cms-sw/cmssw/pull/39412, we should be good for the CMSSW_12_5 release for L1T production. Thanks.

File: L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc [hunk 1]
```diff
@@ -220,7 +220,7 @@ void HPSPFTauProducer::fillDescriptions(edm::ConfigurationDescriptions& descript
   desc.add<double>("minSeedJetPt", 30.0);
   desc.add<double>("maxChargedRelIso", 1.0);
   desc.add<double>("minSeedChargedPFCandPt", 5.0);
-  desc.add<edm::InputTag>("srcL1PFCands", edm::InputTag("l1ctLayer1", "PF"));
+  desc.add<edm::InputTag>("srcL1PFCands", edm::InputTag("l1tLayer1", "PF"));
```
Comment (perrotta) [inline L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc:223]: Is this a bug, or the bug is in the master instead?

**Comment by perrotta** [comment]
@cecilecaillol besides https://github.com/cms-sw/cmssw/pull/39451#pullrequestreview-1134008220 (which must be clarified) this is a verbatim backport of https://github.com/cms-sw/cmssw/pull/39401 and https://github.com/cms-sw/cmssw/pull/39499
Once that gets fixed, this PR can be merged in 12_5. 

Before merging, please also provide an estimate of the increase in the event size, as it was already requested long time ago. A gross estimate is enough, and it should be rather simple to derive it starting from the data samples you have at hand.

**Comment by rappoccio** [comment]
@cecilecaillol ping on this?

**Comment by srimanob** [comment]
Regarding the size, measure with TTbar PU200
`/eos/cms/store/group/offcomp_upgrade-sw/srimanob/Phase2Fall22L1T/outputs/step3-PU200.root`

The collections (listed belowed) add 12776.2 Bytes/Event (Average Compressed Size)
```
l1tEtSums_l1tPhase1JetProducer9x9_UncalibratedPhase1L1TJetFromPfCandidatesMET_HLT. 830.2 637.6
l1tEtSums_l1tPhase1JetSumsProducer9x9_Sums_HLT. 732.2 539.6
l1tEtSums_l1tTrackerEmuEtMiss_L1TrackerEmuEtMiss_HLT. 639 563.6
l1tEtSums_l1tTrackerEmuHTMissExtended_L1TrackerEmuHTMissExtended_HLT. 683.8 607.6
l1tEtSums_l1tTrackerEmuHTMiss_L1TrackerEmuHTMiss_HLT. 639 563.6
l1tHPSPFTaus_l1tHPSPFTauProducerPF__HLT. 11314.2 1740.6
l1tHPSPFTaus_l1tHPSPFTauProducerPuppi__HLT. 6303 1260.4
l1tTkEtMisss_l1tTrackerEtMiss_l1tTrackerEtMiss_HLT. 699.8 570.8
l1tTkHTMisss_l1tTrackerHTMiss_L1TrackerHTMiss_HLT. 689 566.4
l1tTkJets_l1tTrackFastJets_L1TrackFastJets_HLT. 2963.2 869.2
l1tTkJets_l1tTrackJetsExtended_L1TrackJetsExtended_HLT. 5361.8 1199.6
l1tTkJets_l1tTrackJets_L1TrackJets_HLT. 2330.4 753.2
l1tTkJetWords_l1tTrackJetsEmulation_L1TrackJets_HLT. 1870.2 691.2
l1tTkJetWords_l1tTrackJetsExtendedEmulation_L1TrackJetsExtended_HLT. 2516.6 786.4
recoCaloJets_l1tPhase1JetCalibrator9x9_Phase1L1TJetFromPfCandidates_HLT. 1817.2 704.8
recoCaloJets_l1tPhase1JetProducer9x9_UncalibratedPhase1L1TJetFromPfCandidates_HLT. 1845.2 721.6
```

File: L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc (hunk 1 above)
Comment (srimanob) [reply L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc:223]: It seems this line will be overwrite by
https://github.com/cms-sw/cmssw/blob/master/L1Trigger/Phase2L1Taus/python/HPSPFTauProducerPF_cfi.py#L5
`srcL1PFCands = "l1tLayer1:PF",`

However, let's @cms-sw/l1-l2 confirms.

Comment (srimanob) [reply L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc:223]: Agree. Let's spend a little time to clean up things. 

In addition, I see few places in master that still use `l1ctLayer1` which I assume them to change to `l1tLayer1`
https://github.com/cms-sw/cmssw/search?q=l1ctLayer1

Comment (cecilecaillol) [reply L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc:223]: This PR is correct. I will make a bug-fix PR to master.

Comment (srimanob) [reply L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc:223]: Thanks @cecilecaillol

Comment (srimanob) [reply L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc:223]: Will the fixes for [L1Trigger/Phase2L1ParticleFlow/plugins/DeregionizerProducer.cc](https://github.com/cms-sw/cmssw/pull/39704/files#diff-188e0d47aea503e77d5e3a6c5f20cf6e1624a4fe386bf786c35016e8aebbe3fb) and L1Trigger/Phase2L1ParticleFlow/test/make_l1ctLayer1_dumpFiles_cfg.py come in separated backport PR?

Comment (cecilecaillol) [reply L1Trigger/Phase2L1Taus/plugins/HPSPFTauProducer.cc:223]: Yes: https://github.com/cms-sw/cmssw/pull/39705

**Comment by perrotta** [comment]
+1
- Thank you @cecilecaillol

**Pull request merged**